File: astrosetupsback/src/main/java/com/whalensoft/astrosetupsback/application/interfaces/AuthService.java
package com.whalensoft.astrosetupsback.application.interfaces;

import com.whalensoft.astrosetupsback.application.dto.auth.AuthResponseDTO;
import com.whalensoft.astrosetupsback.application.dto.auth.CurrentUserDTO;
import com.whalensoft.astrosetupsback.application.dto.auth.LoginRequestDTO;
import com.whalensoft.astrosetupsback.application.dto.auth.RegisterRequestDTO;

public interface AuthService {

    AuthResponseDTO register(RegisterRequestDTO registerRequestDTO);

    AuthResponseDTO login(LoginRequestDTO loginRequestDTO);

    CurrentUserDTO getCurrentUser(Long userId);
}

File: astroSetupsFrontend/src/services/auth.service.ts
import { apiClient } from '@/api/client';
import type {
  LoginRequest,
  RegisterRequest,
  AuthResponse,
  CurrentUser,
} from '@/interfaces/auth/auth.interface';

export const authService = {
  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>('/auth/login', data);
    return response.data;
  },

  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>('/auth/register', data);
    return response.data;
  },

  async getCurrentUser(): Promise<CurrentUser> {
    const response = await apiClient.get<CurrentUser>('/auth/me');
    return response.data;
  },
};
